src/query/graphite/native/builtin_functions.go

```diff
@@ -1143,6 +1143,25 @@ func exclude(_ *common.Context, input singlePathSpec, pattern string) (ts.Series
 	return r, nil
 }
 
+// pow takes one metric or a wildcard seriesList followed by a constant,
+// and raises the datapoint by the power of the constant provided at each point
+// nolint: gocritic
+func pow(ctx *common.Context, input singlePathSpec, factor float64) (ts.SeriesList, error) {
+	results := make([]*ts.Series, 0, len(input.Values))
+
+	numSteps := input.Values[0].Len()
+	millisPerStep := input.Values[0].MillisPerStep()
+	vals := ts.NewValues(ctx, millisPerStep, numSteps)
+	for i := 0; i < numSteps; i++ {
+		vals.SetValueAt(i, math.Pow(input.Values[0].ValueAt(i), factor))
+	}
+	newName := fmt.Sprintf("pow(%s, %f)", input.Values[0].Name(), factor)
+	results = append(results, ts.NewSeries(ctx, newName, input.Values[0].StartTime(), vals))
```

Comment:
You'll want to do this per series, since this is only performing `pow` to the first series (`input.Values[0]` is just the very first time series in the input).

e.g. you'll want to do this per series like the Graphite implementation:
https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/functions.py#L1650-L1656

So something like `for _, series := range input.Values { numSteps := series.Len(); ...; results = append(results, ts.NewSeries(ctx, newName, series.StartTime(), vals); }`.

Reply:
Just pushed the changes, let me know what you think. Thanks!

Reply:
Actually its failing some checks now I have to go see what I missed